x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/index.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo } from 'react';
+import { ChoroplethMap } from './ChoroplethMap';
+import { ColorProgressionBar } from './ColorProgressionBar';
+import { useAvgDurationByCountry } from '../../../../../hooks/useAvgDurationByCountry';
+
+function getProgressionColor(
+  scale: number,
+  // default to euiColorPrimary (#006BB4)
+  color = { hue: 204, saturation: 100, lightness: { min: 20, max: 90 } }
+) {
+  const { min, max } = color.lightness;
+  const lightness = Math.round(min + (max - min) * (1 - scale));
+  return `hsl(${color.hue},${color.saturation}%,${lightness}%)`;
+}
+
+const RegionMapChartToolTip: React.SFC<{
```

Comment:
@sqren pointed out that there are some issues with the `React.(S)FC` type (implicit typing for children, default props and some other stuff too), and that it's often better to just type `Props`. Can't find the comment immediately though.

Reply:
I'm not strongly against SFC, but since it has a slightly more clunky syntax and comes with a few caveats I avoid it. But if you prefer it I'm fine with that too :)

At some point I wondered whether other agreed with me on this and found this little write-up: https://github.com/typescript-cheatsheets/react-typescript-cheatsheet#function-components (section: `What about `React.FC`/`React.FunctionComponent`?`)

Again: not deal-breaker at all. But that was my reasoning :)